src/gss_names.c

```diff
@@ -281,6 +281,84 @@ uint32_t gssntlm_import_name(uint32_t *minor_status,
                                        output_name);
 }
 
+size_t gssntlm_get_attrs_count(const struct gssntlm_name_attribute *attrs)
+{
+    size_t attrs_count = 0;
+    if (!attrs) {
+        return 0;
+    }
+    while (attrs[attrs_count].attr_name != NULL) {
+        attrs_count++;
+    }
+
+    return attrs_count;
+}
+
+int gssntlm_copy_attrs(const struct gssntlm_name_attribute *src,
+                       struct gssntlm_name_attribute **dst)
+{
+    struct gssntlm_name_attribute *copied_attrs;
+    size_t attrs_count = gssntlm_get_attrs_count(src);
+
+    if (attrs_count == 0) {
+        *dst = NULL;
+        return 0;
+    }
+
+    copied_attrs = calloc((attrs_count + 1), /* +1 for terminator entry */
+                          sizeof(struct gssntlm_name_attribute));
+    if (copied_attrs == NULL) {
+        return ENOMEM;
+    }
+
+    for (size_t i = 0; i < attrs_count; i++) {
+        /* read-only persistent data ptr can be just copied */
+        copied_attrs[i].attr_name = src[i].attr_name;
+        copied_attrs[i].attr_value.length = src[i].attr_value.length;
+        copied_attrs[i].attr_value.value = malloc(src[i].attr_value.length);
+        if (copied_attrs[i].attr_value.value == NULL) {
+            gssntlm_release_attrs(&copied_attrs);
+            return ENOMEM;
+        }
+        memcpy(copied_attrs[i].attr_value.value, src[i].attr_value.value,
+               src[i].attr_value.length);
+    }
+    /* terminator entry is filled with zeroes by calloc */
+
+    *dst = copied_attrs;
+    return 0;
+}
+
+struct gssntlm_name_attribute *
+    gssntlm_find_attr(struct gssntlm_name_attribute *attrs,
```

Comment:
Spurious indentation.

Reply:
please SHOW ME how to do it properly. Here we have contradiction of your indentation rules and 'line length limit of 80 chars' rule. In either case you can say 'you have done wrong'.

Please add such an example to https://github.com/gssapi/gss-ntlmssp/wiki/Contributions#indentation - here single parameter definition is too long to fit in 80 characters.

It's NOT better:
```
struct gssntlm_name_attribute *gssntlm_find_attr(
    struct gssntlm_name_attribute *attrs,
    const char *attr_name, size_t attr_name_len)
```

And this is TOO LONG:
```
struct gssntlm_name_attribute *gssntlm_find_attr(struct gssntlm_name_attribute *attrs,
                                                 const char *attr_name,
                                                 size_t attr_name_len)
```

Reply:
One of the rare cases where indentation is hard, sorry it happened to you.
Pkease use this:
```
struct gssntlm_name_attribute *gssntlm_find_attr(
                                        struct gssntlm_name_attribute *attrs,
                                        const char *attr_name,
                                        size_t attr_name_len)
```
(Indent the first argument as far right as you can (but always only by 4 space increments) without breaching the 80 columns rule.

Reply:
Aha, I got your point, will fix all the places like that.
P.S. If you can add this explanation to your page https://github.com/gssapi/gss-ntlmssp/wiki/Contributions#indentation 
then you can refer to it later every time someone else has this problem :-)

Reply:
Added there, let me know if it is understandable.

Reply:
Great, thanks!